store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -964,9 +964,11 @@ public void run() {
             boolean result = false;
             for (int i = 0; i < RETRY_TIMES_OVER && !result; i++) {
```

Comment:
right, that's what I also thought and ask above about `RETRY_TIMES_OVER`. Thank you for clarifying this.

How about this (flushing until data is available and `MAX_FLUSH_NUM` (which can be quite large) is reached, and log warnings when cannot flush)?

```java
for (int i = 0; i < MAX_FLUSH_NUM && remainHowManyDataToFlush() > 0; i++) {
     if (!CommitLog.this.mappedFileQueue.flush(0))
          CommitLog.log.warn(this.getServiceName() + " service shutdown, failed flush retried "
                        + (i + 1) + " times");
}
CommitLog.log.info(this.getServiceName() + " service shutdown, all data is flushed successfully");
```